bundles/org.openhab.binding.bluelink/src/main/java/org/openhab/binding/bluelink/internal/handler/BluelinkVehicleHandler.java

```diff
@@ -0,0 +1,496 @@
+            updateState(GROUP_WARNINGS, CHANNEL_TIRE_PRESSURE_WARNING_RR,
+                    OnOffType.from(tirePressureWarning.rearRight() > 0));
+            updateState(GROUP_WARNINGS, CHANNEL_TIRE_PRESSURE_WARNING_RL,
+                    OnOffType.from(tirePressureWarning.rearLeft() > 0));
+        }
+
+        // Location
+        final var location = data.vehicleLocation();
+        if (location != null && location.coord() != null) {
+            updateState(GROUP_STATUS, CHANNEL_LOCATION, new PointType(new DecimalType(location.coord().latitude()),
+                    new DecimalType(location.coord().longitude()), new DecimalType(location.coord().altitude())));
+        }
+    }
+
+    private void updateState(final String group, final String channel, final State state) {
+        updateState(new ChannelUID(getThing().getUID(), group, channel), state);
+    }
+
+    private OpenClosedType toOpenClosed(final int value) {
+        return value > 0 ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+    }
+
+    private OpenClosedType toOpenClosed(final boolean value) {
+        return value ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+    }
+
+    @Nullable
+    public BluelinkApi getApi() {
+        final Bridge bridge = getBridge();
+        if (bridge == null) {
+            return null;
+        }
+        final BluelinkAccountHandler handler = (BluelinkAccountHandler) bridge.getHandler();
+        return handler != null ? handler.getApi() : null;
+    }
+
+    private void createDynamicChannels() {
+        final List<Channel> newChannels;
+        final String group;
+        if (isElectric) {
+            group = GROUP_CHARGING;
+            newChannels = List.of(
+                    buildChannel(GROUP_CHARGING, CHANNEL_EV_BATTERY_SOC, "Number:Dimensionless", "ev-soc"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_EV_CHARGING, "Switch", "ev-charging"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_EV_PLUGGED_IN, "Switch", "ev-plugged-in"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_CHARGE_LIMIT_DC, "Number:Dimensionless", "charge-limit"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_CHARGE_LIMIT_AC, "Number:Dimensionless", "charge-limit"),
+
+                    buildChannel(GROUP_CHARGING, CHANNEL_TIME_TO_FULL_CURRENT, "Number:Time", "charge-time"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_TIME_TO_FULL_FAST, "Number:Time", "charge-time"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_TIME_TO_FULL_PORTABLE, "Number:Time", "charge-time"),
+                    buildChannel(GROUP_CHARGING, CHANNEL_TIME_TO_FULL_STATION, "Number:Time", "charge-time"));
+        } else {
+            group = GROUP_FUEL;
+            newChannels = List.of(buildChannel(GROUP_FUEL, CHANNEL_FUEL_LEVEL, "Number:Dimensionless", "fuel-level"),
+                    buildChannel(GROUP_FUEL, CHANNEL_FUEL_LOW_WARNING, "Switch", "warning"));
```

Comment:
Please use bindingconstants for channel id's and where possible ItemType constants from core